coindecko/pyscript/ingest_data.py

```diff
@@ -1,49 +1,47 @@
 import requests
 import pandas as pd
 
-## Waiting until thingss break
-per_page=250
+# Waiting until things break
+per_page = 250
 page = 1
 json_df = []
 
-### 
+# Write a function to test if the first id is bitcoin, if it's true the convert the data to a dataframe
+
+
+def test_data_ingested(df):
+    if df[0]['id'] == 'bitcoin':
+        df = pd.DataFrame(df, columns=['id', 'name', 'current_price', 'total_volume', 'market_cap', 'last_updated'])
+        return df
+    else:
+        print('Wrong data imported! Kindly check the data source')
+
+
 while True:
     print("----")
     url = f"https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&per_page={per_page}&page={page}"
     print("Requesting", url)
-    ## Do the HTTP get request
+    # Do the HTTP get request
     response = requests.get(url)
     # Code here will only run if the request is successful
-    if (response.status_code == 200):
+    if response.status_code == 200:
         print('The request was a success!')
     # Code here will react to failed requests
     else:
-        print('There is a failed request! ', response.status_code)   
-    # Decode the JSON response into a dictionary and use the data
+        print('There is a failed request! ', response.status_code)
+        # Decode the JSON response into a dictionary and use the data
     data = response.json()
     # Debug by printing out the first data extracted
-    try:
-        print(data[0])
```

Comment:
Better to use a breakpoint in a debugger here with the print statement than a try-catch -- a print statement function is not going to throw an exception

Reply:
Do you mean that red dot at the side of the editor? I never understood how that works. Maybe, you could show me.